models/groups.go

```diff
@@ -136,3 +143,63 @@ func (g *Group) BuildSuccessfulJsonResponse(ctx context.Context) ([]byte, error)
 	}
 	return jsonResponse, nil
 }
+
+type CreateGroup struct {
+	Description *string `json:"description"`
+	Precedence *int64 `json:"precedence"`
+	GroupName string
+}
+
+func (g *CreateGroup) ValidateCreateGroupRequest(ctx context.Context) []error {
+	var validationErrs []error
+
+	if g.Description == nil {
+		validationErrs = append(validationErrs, NewValidationError(ctx, InvalidGroupDescription, MissingGroupDescription))
+	} else {
```

Comment:
Could this be simplified to this

} else if m, _ := regexp.MatchString("(?i)^role_.*", *g.Description); m {